### PR TITLE
[flake8-torch] Add codemod

### PR DESCRIPTION
**Pull request by kit1980**
### <samp>🤖 Generated by Copilot at 5c56c4a</samp>

Add a codemod feature to the flake8-torch plugin that can automatically fix some lint violations related to deprecated or removed functions in PyTorch. Update the flake8-torch checker and tests to support the codemod feature. Add new test fixtures for the codemod feature. Rename some existing test fixtures for clarity.

### CODE REVIEW

File: tools/flake8-torch/tests/test_flake8_torch.py [hunk 1]
```diff
@@ -1,28 +1,46 @@
...
-            assert _results(source.readlines()) == expected_results
+            assert _checker_results(source.readlines()) == expected_results
+
+
+def test_codemod_fixtures():
+    for source_path in (FIXTURES_PATH / "codemod").glob("*.py"):
+        LOGGER.info("Testing %s", source_path.relative_to(Path.cwd()))
+        expected_path = str(source_path) + ".out"
+        with open(expected_path) as expected:
+            expected_results = expected.read()
```
Comment (malfet) [inline tools/flake8-torch/tests/test_flake8_torch.py:45]: Nit(use `Path` features)
```suggestion
        expected_path = source_path.with_suffix(".py.out")
        expected_results = expected_path.read_text()
```

File: tools/flake8-torch/flake8_torch/checker.py [hunk 2]
```diff
@@ -1,19 +1,52 @@
...
+import contextlib
+import io
+import sys
 
 __version__ = "0.0.1"
 
+CYAN = "\033[96m"
+RED = "\033[31m"
+BOLD = "\033[1m"
+ENDC = "\033[0m"
```
Comment (malfet) [inline tools/flake8-torch/flake8_torch/checker.py:17]: Don't emit tty escape codes to pipes or on Windows
```suggestion
IS_TTY = hasattr(sys.stdout, "isatty") and sys.stdout.isatty()
CYAN = "\033[96m" if IS_TTY else ""
RED = "\033[31m" if IS_TTY else ""
BOLD = "\033[1m" if IS_TTY else ""
ENDC = "\033[0m" if IS_TTY else ""
```

File: tools/flake8-torch/flake8_torch/checker.py [hunk 3]
```diff
@@ -42,27 +102,110 @@ class TorchChecker:
...
+                new_module = new_module.deep_replace(
+                    violation.node, violation.replacement
+                )
+                fixes_count += 1
+            try:
+                path = Path(self.context.filename).relative_to(Path.cwd())
+            except ValueError:
+                # Not a subpath of a current dir, use absolute path
+                path = self.context.filename
+            print(f"{path}{violation.codemod_result()}")
```
Comment (malfet) [inline tools/flake8-torch/flake8_torch/checker.py:144]: I guess it's a debug print?
```suggestion
```

File: tools/flake8-torch/flake8_torch/checker.py [hunk 4]
```diff
@@ -42,27 +102,110 @@ class TorchChecker:
...
+        if fixes_count == 0:
+            raise codemod.SkipFile("No changes")
+        return new_module
+
+
+def main():
```
Comment (malfet) [inline tools/flake8-torch/flake8_torch/checker.py:150]:
```suggestion
def main() -> None:
```

File: tools/flake8-torch/flake8_torch/checker.py [hunk 5]
```diff
@@ -42,27 +102,110 @@ class TorchChecker:
...
+                files,
+                jobs=args.jobs,
+                unified_diff=(None if args.fix else DIFF_CONTEXT),
+                hide_progress=True,
+                format_code=False,
+                repo_root=None,
+            )
+    except KeyboardInterrupt:
+        print("Interrupted!", file=sys.stderr)
+        return 2
```
Comment (malfet) [inline tools/flake8-torch/flake8_torch/checker.py:191]: `main` return code is ignored, isn't it?
```suggestion
        sys.exit(2)
```

File: tools/flake8-torch/flake8_torch/checker.py [hunk 6]
```diff
@@ -42,27 +102,110 @@ class TorchChecker:
...
+    if result.successes > 0:
+        if args.fix:
+            print(f"Transformed {result.successes} files successfully.")
+        else:
+            print(
+                f"[{CYAN}*{ENDC}] {result.successes} "
+                "potentially fixable with the --fix option"
+            )
+
+    return 1 if result.failures > 0 else 0
```
Comment (malfet) [inline tools/flake8-torch/flake8_torch/checker.py:207]:
```suggestion
    if result.failures > 0:
        sys.exit(1)
```

**Review by malfet — APPROVED** [review]

File: tools/flake8-torch/flake8_torch/checker.py (hunk 3 above)
Comment (kit1980) [reply tools/flake8-torch/flake8_torch/checker.py:144]: No, it's the output of the checker when it runs in standalone mode (not as a flake8 plugin).

File: tools/flake8-torch/flake8_torch/checker.py (hunk 5 above)
Comment (kit1980) [reply tools/flake8-torch/flake8_torch/checker.py:191]: Right, I'll fix all of these.

**Pull request merged**